custom-elements/reactions/HTMLSourceElement.html

```diff
@@ -0,0 +1,187 @@
+<!DOCTYPE html>
```

Comment:
Since this file introduces new helpers, can you separate it out into a separate PR, so that someone can review the helpers in more detail?